[Unit Test]TestCreateSSHShell fails

<!-- Please use this template while reporting an issue, providing as much information as possible. Failure to do so may result in a delayed response. Thank you! -->

**The exact command to reproduce the issue**:

Unit test `TestCreateSSHShell` fails with following output.

```
$ make test
Stopping "mock-machine-0.15651925"...
Machine "mock-machine-0.15651925" was stopped.
Stopping "mock-machine-0.21426387"...
Machine "mock-machine-0.21426387" was stopped.
Using SSH client type: external
&{[-F /dev/null -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none -o LogLevel=quiet -o PasswordAuthentication=no -o ServerAliveInterval=60 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null root@localhost -p 52127] /usr/bin/ssh <nil>}
/usr/bin/ssh -F /dev/null -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none -o LogLevel=quiet -o PasswordAuthentication=no -o ServerAliveInterval=60 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null root@localhost -p 52127 exit
--- FAIL: TestCreateSSHShell (0.16s)
    ssh_mock.go:176: Listening on 127.0.0.1:52127
    api_mock.go:62: MockAPI.Load: &{ConfigVersion:0 Driver:0xc00042e000 DriverName: HostOptions:<nil> Name: RawDriver:[]} - <nil>
    driver_mock.go:47: MockDriver.GetState: Running
    ssh_mock.go:83: Accepting...
    ssh_mock.go:83: Accepting...
    ssh_mock.go:135: exec request received: &{Type:exec WantReply:true Payload:[0 0 0 4 101 120 105 116] ch:0xc000180840 mux:<nil>}
    ssh_mock.go:154: SendRequest failed: EOF
    cluster_test.go:438: Error running ssh command: exit status 255
FAIL
```

I found this error has happend from v1.7.3 which changed directory(`pkg/minikube/machine`, `pkg/minikube/cluster`).

But, when I checkout recent PR's unit test result, I could not find same error.
So, I think this issue may happen when local machine's OS is macOS or specific OS.

**The full output of the command that failed**:<details>

```
$ make test
./test.sh
= make lint =============================================================
ok
= go mod ================================================================
ok
= boilerplate ===========================================================
ok
= schema_check ==========================================================
ok
= go test ===============================================================
ok      k8s.io/minikube/cmd/minikube/cmd        3.606s  coverage: 19.3% of statements
ok      k8s.io/minikube/cmd/minikube/cmd/config 0.364s  coverage: 19.6% of statements
ok      k8s.io/minikube/pkg/addons      0.100s  coverage: 50.5% of statements
ok      k8s.io/minikube/pkg/drivers     0.032s  coverage: 19.6% of statements
ok      k8s.io/minikube/pkg/drivers/hyperkit    0.032s  coverage: 13.8% of statements
ok      k8s.io/minikube/pkg/minikube/bootstrapper       2.443s  coverage: 75.7% of statements
ok      k8s.io/minikube/pkg/minikube/bootstrapper/bsutil        0.137s  coverage: 62.8% of statements
ok      k8s.io/minikube/pkg/minikube/bootstrapper/bsutil/ktmpl  0.017s  coverage: 100.0% of statements
ok      k8s.io/minikube/pkg/minikube/bootstrapper/images        0.010s  coverage: 98.0% of statements
ok      k8s.io/minikube/pkg/minikube/cluster    0.054s  coverage: 23.2% of statements
ok      k8s.io/minikube/pkg/minikube/command    0.033s  coverage: 4.7% of statements
ok      k8s.io/minikube/pkg/minikube/config     1.046s  coverage: 69.1% of statements
ok      k8s.io/minikube/pkg/minikube/cruntime   0.058s  coverage: 50.9% of statements
ok      k8s.io/minikube/pkg/minikube/download   44.342s coverage: 21.6% of statements
ok      k8s.io/minikube/pkg/minikube/driver     0.043s  coverage: 38.7% of statements
ok      k8s.io/minikube/pkg/minikube/extract    0.037s  coverage: 56.9% of statements
ok      k8s.io/minikube/pkg/minikube/kubeconfig 0.059s  coverage: 77.3% of statements
ok      k8s.io/minikube/pkg/minikube/localpath  0.016s  coverage: 20.4% of statements
ok      k8s.io/minikube/pkg/minikube/logs       0.044s  coverage: 1.3% of statements
Stopping "mock-machine-0.15651925"...
Machine "mock-machine-0.15651925" was stopped.
Stopping "mock-machine-0.21426387"...
Machine "mock-machine-0.21426387" was stopped.
Using SSH client type: external
&{[-F /dev/null -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none -o LogLevel=quiet -o PasswordAuthentication=no -o ServerAliveInterval=60 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null root@localhost -p 52127] /usr/bin/ssh <nil>}
/usr/bin/ssh -F /dev/null -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none -o LogLevel=quiet -o PasswordAuthentication=no -o ServerAliveInterval=60 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null root@localhost -p 52127 exit
--- FAIL: TestCreateSSHShell (0.16s)
    ssh_mock.go:176: Listening on 127.0.0.1:52127
    api_mock.go:62: MockAPI.Load: &{ConfigVersion:0 Driver:0xc00042e000 DriverName: HostOptions:<nil> Name: RawDriver:[]} - <nil>
    driver_mock.go:47: MockDriver.GetState: Running
    ssh_mock.go:83: Accepting...
    ssh_mock.go:83: Accepting...
    ssh_mock.go:135: exec request received: &{Type:exec WantReply:true Payload:[0 0 0 4 101 120 105 116] ch:0xc000180840 mux:<nil>}
    ssh_mock.go:154: SendRequest failed: EOF
    cluster_test.go:438: Error running ssh command: exit status 255
FAIL
coverage: 41.9% of statements
FAIL    k8s.io/minikube/pkg/minikube/machine    79.174s
ok      k8s.io/minikube/pkg/minikube/notify     3.480s  coverage: 92.9% of statements
ok      k8s.io/minikube/pkg/minikube/out        0.033s  coverage: 80.0% of statements
ok      k8s.io/minikube/pkg/minikube/perf       0.009s  coverage: 54.5% of statements
ok      k8s.io/minikube/pkg/minikube/problem    0.016s  coverage: 97.1% of statements
ok      k8s.io/minikube/pkg/minikube/proxy      0.026s  coverage: 100.0% of statements
ok      k8s.io/minikube/pkg/minikube/registry   0.042s  coverage: 90.7% of statements
ok      k8s.io/minikube/pkg/minikube/registry/drvs/hyperkit     0.037s  coverage: 41.5% of statements
ok      k8s.io/minikube/pkg/minikube/service    2.590s  coverage: 86.2% of statements
ok      k8s.io/minikube/pkg/minikube/sshutil    0.506s  coverage: 100.0% of statements
ok      k8s.io/minikube/pkg/minikube/storageclass       0.030s  coverage: 97.2% of statements
ok      k8s.io/minikube/pkg/minikube/translate  0.010s  coverage: 11.4% of statements
ok      k8s.io/minikube/pkg/minikube/tunnel     1.629s  coverage: 58.8% of statements
ok      k8s.io/minikube/pkg/util        0.619s  coverage: 65.7% of statements
ok      k8s.io/minikube/pkg/util/lock   0.011s  coverage: 18.2% of statements
ok      k8s.io/minikube/pkg/util/retry  0.007s  coverage: 0.0% of statements
FAIL
make: *** [test] Error 32
```

</details>

**The output of the `minikube logs` command**:

Nothing log due to test.

**The operating system version**:

OS: macOS Mojave 10.14.5
Go: go version go1.13.7 darwin/amd64
Minikube: master(commit: e6ba335e9c5e5d7e37065e00d67cb9340b65a713)

I'll send a PR to fix this issue.

--- a/pkg/minikube/machine/cluster_test.go
+++ b/pkg/minikube/machine/cluster_test.go
@@ -413,7 +413,7 @@ func TestGetHostStatus(t *testing.T) {
 
 func TestCreateSSHShell(t *testing.T) {
 	api := tests.NewMockAPI(t)
-	// Setting the default client to native gives much better performance.
+	// Setting the default ssh client to native for test stability.
 	ssh.SetDefaultClient(ssh.Native)
 
 	s, _ := tests.NewSSHServer(t)

--- a/pkg/minikube/machine/cluster_test.go
+++ b/pkg/minikube/machine/cluster_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/docker/machine/libmachine/drivers"
 	"github.com/docker/machine/libmachine/host"
 	"github.com/docker/machine/libmachine/provision"
+	"github.com/docker/machine/libmachine/ssh"
 	"github.com/docker/machine/libmachine/state"
 	"github.com/spf13/viper"
 	"k8s.io/minikube/pkg/minikube/config"
@@ -412,6 +413,8 @@ func TestGetHostStatus(t *testing.T) {
 
 func TestCreateSSHShell(t *testing.T) {
 	api := tests.NewMockAPI(t)
+	// Setting the default client to native gives much better performance.
+	ssh.SetDefaultClient(ssh.Native)
 
 	s, _ := tests.NewSSHServer(t)
 	port, err := s.Start()

